PyExpUtils/utils/dict.py

```diff
@@ -54,13 +85,46 @@ def get(d: Any, key: str, default: Any = None) -> Any:
     return el
 
 def equal(d1: Dict[T, Any], d2: Dict[T, Any], ignore: Sequence[T] = []):
```

Comment:
These are 3 different types of equality checks for dictionaries. I should think more about a way to compress these ideas into a more unified language / way to check for dictionary equality.